File: Utt2Database/ServerMatch.vb
Imports System
Imports System.Collections.Generic
Imports Microsoft.VisualBasic

Namespace Utt2Database
    Partial Public Class ServerMatch
        Public Property Id As Long

        Public Property ServerId As Long

        Public Property StartTime As Date

        Public Property MapName As String

        Public Property ServerPlayeridCounter As Integer?

        Public Overridable Property PlayerLogs As ICollection(Of PlayerLog) = New List(Of PlayerLog)()

        Public Overridable Property PlayerStats As ICollection(Of PlayerStat) = New List(Of PlayerStat)()

        Public Overridable Property Server As Server
    End Class
End Namespace

File: src/IndexerSharp/Utt2Database/ServerMatch.cs
using System.Xml.Linq;

namespace Naomai.UTT.Indexer.Utt2Database;
public partial class ServerMatch
{
    public long? Id { get; set; } 
    public long ServerId { get; set; } 
    public DateTime StartTime { get; set; }
    public string MapName { get; set; } = null!;
    public int? ServerPlayeridCounter { get; set; }

    public virtual ICollection<PlayerLog> PlayerLogs { get; set; } = new List<PlayerLog>();
    public virtual ICollection<PlayerStat> PlayerStats { get; set; } = new List<PlayerStat>();
    public Server Server = null!;

    public override string ToString()
    {
        return "M" + Id + "#" + Server.AddressGame + "[" + MapName + "]";
    }
}